### ISSUE TITLE
wrong_self_convention false positive for async fn `into_*`

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust-clippy#4037]
```rust
#![feature(async_await)]

pub struct Foo;
pub struct Bar;

impl Foo {
    pub async fn into_bar(self) -> Bar {
        Bar
    }
}
```

([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=4e8a9c22a0ceb7283bc7c3ba07277540)) gives

```
warning: methods called `into_*` usually take self by value; consider choosing a less ambiguous name
 --> src/lib.rs:7:27
  |
7 |     pub async fn into_bar(self) -> Bar {
  |                           ^^^^
  |
  = note: #[warn(clippy::wrong_self_convention)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

    Finished dev [unoptimized + debuginfo] target(s) in 0.71s
```

**Comment by Nemo157** [comment]
I haven't checked whether there are issues with the other patterns that `wrong_self_convention` is looking for, just noticed this one when running Clippy on a project.

**Comment by JohnTitor** [comment]
It seems this FP is no longer reproduced in playground. Could you confirm?

**Comment by Nemo157** [comment]
Yep, looks like it's fixed. Probably worth leaving open (and tagging easy) to add `async fn` testcases to the UI tests to make sure it doesn't regress.

**Comment by JohnTitor** [comment]
Sounds good! Marked as "good first issue".
